manager/state/raft/storage.go

```diff
@@ -231,10 +234,8 @@ func (n *Node) restoreFromSnapshot(data []byte, forceNewCluster bool) error {
 	oldMembers := n.cluster.Members()
 
 	if !forceNewCluster {
+		membersCallback(snapshot.Membership.Members)
```

Comment:
Could this just return `snapshot.Membership` instead of invoking a callback and doing part of the work itself? It feels really weird that it depends on `membersCallback` to add members, but it removes old members directly.

Also, if we keep the callback, we should check the error it returns.